### ISSUE TITLE
u.ABmag.to(u.Jy) should not work

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#9081]
Update: Desired outcome is for `u.ABmag.to(u.Jy)` to throw exception.

---

At the moment, the following returns a scalar value:

```
In [11]: u.ABmag.to(u.Jy)                                                                                                                              
Out[11]: 1445.4397707459234
```

However, this is not a linear conversion, so I think it's misleading for this to work?

**Comment by abhijeetmanhas** [comment]
@astrofrog do you mean that there should be spectral density as a second parameter for above conversion? If yes, I will like to work on this issue.

**Comment by abhijeetmanhas** [comment]
@astrofrog output be like this?
`<Quantity 1445.4397707459234. Jy>`

**Comment by astrofrog** [comment]
No, the issue is that to me this is confusing because the conversion from ABmag to Jy is not a simple scaling, so it doesn't make sense to return a scalar value. But @mhvk will need to confirm whether this is intentional.

**Comment by mhvk** [comment]
Currently, `MagUnit.to` behaves exactly the same as `UnitBase.to`, i.e., it takes a `value` argument with a default of `1`. This is not a very useful default if one is converting to a physical value, and I agree that for that case we should raise an error instead (but perhaps leaving log-unit to log-unit intact? Also fine to just insist on an actual value).

**Comment by astrofrog** [comment]
Yes I think the cleanest would be to require a value when doing a log to physical conversion or vice-versa and leave it unchanged otherwise.

**Comment by dhomeier** [comment]
Intuitively it would make more sense to me to have `0` as default `value` argument for `MagUnit` (and log-units in general), although their `__mul__` method looks perhaps non-intuitive at first.
So silently adopting a default might be a bit dangerous either way.
```
>>> (0*u.ABmag).to(u.Jy)
<Quantity 3630.7805477 Jy>
>>> (5*u.ABmag).to(u.Jy)
<Quantity 36.30780548 Jy>
>>> (10*u.ABmag).to(u.Jy)
<Quantity 0.36307805 Jy>
```

**Comment by pllim** [comment]
Ultimately people should stop using magnitude system. 🔥 

But since we cannot stop them, I think we should just close this as won't fix. Having special default just for magnitude is also confusing. I don't think this behavior affects most user that uses Quantity.

**Comment by astrofrog** [comment]
Well @mhvk did agree above that we could maybe adjust ``MagUnit.to`` to behave better? Currently this could lead to actual user issues if they think they can trust this.

**Comment by pllim** [comment]
I was only here because of the `needs-reconfirmation` label. Do you want me to re-open and accept the suggestion to throw error?

**Comment by astrofrog** [comment]
Yes I personally think we should keep open
